pom.xml

```diff
@@ -526,5 +604,49 @@
                 <skipTests>true</skipTests>
             </properties>
         </profile>
+        <profile>
+            <id>release</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-gpg-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>sign-artifacts</id>
+                                <phase>verify</phase>
+                                <goals>
+                                    <goal>sign</goal>
+                                </goals>
+                                <configuration>
+                                    <!--suppress UnresolvedMavenProperty -->
+                                    <keyname>${releaseSigningKey}</keyname>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-enforcer-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>enforce-property</id>
+                                <goals>
+                                    <goal>enforce</goal>
+                                </goals>
+                                <configuration>
+                                    <rules>
+                                        <requireProperty>
+                                            <property>releaseSigningKey</property>
+                                            <message>You must specify your GPG key when running a release</message>
+                                        </requireProperty>
```

Comment:
suggest adding:

```
<requireReleaseDeps>
    <message>No Snapshots Allowed!</message>
</requireReleaseDeps>
```

Reply:
alternatively:

[banDynamicVersions](https://maven.apache.org/enforcer/enforcer-rules/banDynamicVersions.html) - bans all dependencies requiring version resolution at build time (i.e. version ranges, placeholders RELEASE/LATEST or SNAPSHOT versions).

Reply:
maybe this too: [requireReleaseVersion](https://maven.apache.org/enforcer/enforcer-rules/requireReleaseVersion.html) - enforces that the artifact is not a snapshot.